gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -544,23 +544,25 @@ void connectHelixManagerWithRetry() {
   private void addInstanceTags() {
     HelixManager receiverManager = getReceiverManager();
     if (receiverManager.isConnected()) {
-      // The helix instance associated with this container should be consistent on helix tag
-      List<String> existedTags = receiverManager.getClusterManagmentTool()
-          .getInstanceConfig(this.clusterName, this.helixInstanceName).getTags();
-      Set<String> desiredTags = new HashSet<>(
-          ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY));
-      if (!desiredTags.isEmpty()) {
-        // Remove tag assignments for the current Helix instance from a previous run
-        for (String tag : existedTags) {
-          if (!desiredTags.contains(tag))
-            receiverManager.getClusterManagmentTool().removeInstanceTag(this.clusterName, this.helixInstanceName, tag);
-          logger.info("Removed unrelated helix tag {} for instance {}", tag, this.helixInstanceName);
+      try {
+        Set<String> desiredTags = new HashSet<>(ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY));
+        if (!desiredTags.isEmpty()) {
```

Comment:
From what I can tell, the reason this works is because the `GobblinYarnTaskRunner` is the only one to set the config key for `HELIX_INSTANCE_TAGS_KEY`. And so by proxy we know that if this desired tags is empty, then we can skip this code because this is not being run by the `GobblinYarnTaskRunner`. But this all feels pretty messy and a bad experience for anyone in the future working on this.

Can we find a better approach with an override method? Isn't the real code smell that this code is inherently coupled with the YarnService / YarnAutoScalingManager? But since this code is being used outside the yarn context, it should instead live inside the `GobblinYarnTaskRunner` somehow.

Admittedly, I am not that big of a fan of inheritance, but since we are already using it in this class it makes sense to use that approach 

```
public class GobblinYarnTaskRunner extends GobblinTaskRunner
```

Reply:
Yes that's a good point, if it is yarn specific it should lie in the `GobblinYarnTaskRunner` perhaps. However, we _can_ technically run gobblin-cluster in _non-standalone_ mode and which would rely on this code. Afaik we don't actually use that and best practice may be to use the child class instead however I don't want to change anything too drastically for the current fix.

Reply:
Does non standalone gobblin mean yarn? Does that mean all usages of yarn don't use the GobblinYarnTaskRunner?

Reply:
Standalone mode = no yarn. Yes, non standalone does use Yarn. From what I can see in the code it changes how the [GobblinHelixMultiManager](https://jarvis.corp.linkedin.com/codesearch/result/?name=GobblinHelixMultiManager.java&path=gobblin-elr%2Fgobblin-cluster%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fgobblin%2Fcluster&reponame=linkedin%2Fgobblin-elr#275) and more importantly [GobblinClusterManager](https://jarvis.corp.linkedin.com/codesearch/result/?name=GobblinClusterManager.java&path=gobblin-elr%2Fgobblin-cluster%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fgobblin%2Fcluster&reponame=linkedin%2Fgobblin-elr#286) work. I have to dig around more to see how `GobblinYarnTaskRunner` is used. Let's investigate offline.